jetty-core/jetty-quic/jetty-quic-api/src/main/java/org/eclipse/jetty/quic/api/frames/TransportParameters.java

```diff
@@ -0,0 +1,182 @@
+     */
+    public static final Id<Long> INITIAL_MAX_STREAM_DATA_BIDIRECTIONAL_REMOTE = new LongId(0x06);
+    /**
+     * <p>The initial unidirectional stream max data.</p>
+     * <p>A local peer sends this parameter to inform the remote peer about
+     * the max data the local peer is willing to receive on unidirectional streams.</p>
+     */
+    public static final Id<Long> INITIAL_MAX_STREAM_DATA_UNIDIRECTIONAL = new LongId(0x07);
+    /**
+     * <p>The initial bidirectional max streams.</p>
+     * <p>A local peer sends this parameter to inform the remote peer about
+     * the max number of streams the local peer is willing to receive.</p>
+     */
+    public static final Id<Long> INITIAL_MAX_STREAMS_BIDIRECTIONAL = new LongId(0x08);
+    /**
+     * <p>The initial unidirectional max streams.</p>
+     * <p>A local peer sends this parameter to inform the remote peer about
+     * the max number of streams the local peer is willing to receive.</p>
+     */
+    public static final Id<Long> INITIAL_MAX_STREAMS_UNIDIRECTIONAL = new LongId(0x09);
+
+    private final Map<Id<?>, Object> parameters;
+
+    public TransportParameters()
+    {
+        this.parameters = new HashMap<>();
+    }
+
+    @Override
+    public Iterator<Map.Entry<Id<?>, Object>> iterator()
+    {
+        return parameters.entrySet().iterator();
+    }
+
+    public <T> T get(Id<T> id)
+    {
+        Object value = parameters.get(id);
+        return id.cast(value);
+    }
+
+    public <T> T put(Id<T> id, T value)
+    {
+        return id.cast(parameters.put(id, value));
+    }
+
+    public <T> T putIfAbsent(Id<T> id, T value)
+    {
+        return id.cast(parameters.putIfAbsent(id, value));
+    }
+
+    public void putGreaseParameter()
+    {
+        ThreadLocalRandom random = ThreadLocalRandom.current();
+        long id = 31 * random.nextLong(MAX_N + 1) + 27;
+        byte[] bytes = new byte[8];
+        random.nextBytes(bytes);
+        put(new BytesId(id), bytes);
+    }
+
+    public int size()
+    {
+        return parameters.size();
+    }
+
+    @Override
+    public String toString()
+    {
+        return "%s@%x[%s]".formatted(getClass().getSimpleName(), hashCode(), parameters);
+    }
+
+    public abstract static class Id<T>
+    {
+        private final long id;
+
+        public Id(long id)
+        {
+            this.id = id;
+        }
+
+        public long getId()
+        {
+            return id;
+        }
+
+        public abstract T cast(Object value);
+
+        @Override
+        public int hashCode()
+        {
+            return Long.hashCode(id);
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            if (this == obj)
+                return true;
+            if (obj instanceof Id<?> that)
+                return id == that.id;
+            return false;
+        }
+
+        @Override
+        public String toString()
+        {
+            return "%d".formatted(id);
```

Comment:
this should be more than just the ID as it is possible to have two instances of the same ID